MCDataProducts/inc/SimParticle.hh

```diff
@@ -125,6 +129,8 @@ namespace mu2e {
                      CLHEP::HepLorentzVector aendMomentum,
                      float                  aendGlobalTime,
                      float                  aendProperTime,
+                     float                  apreLastStepGlobalTime,
```

Comment:
I am worried about inflating the size of general-purpose objects, used in great numbers, to support specialty use cases.  Can you please explain why this information is of such general interest that it needs to be stored for every single SimParticle?

Reply:
Let's discuss this following Krzysztof's presentation this afternoon.

Reply:
Note that this is per particle, not per step though.
I suggest we discuss it at our meeting. Meanwhile, do you see an alternative approach?

Reply:
Here's the payload breakdown for a typical mixing input from MDC2020: 63% comes from SimParticle.  This size causes a real limitation on mixing (cosmic mixing takes >8GB memory), though we use extremely little of the payload.
Details for branch: Events
Number of entries:  91192
        Size     Size/Entry   Fraction  Data Product Name
     (bytes)       (bytes)

   193504968         2122.0      0.633  mu2e::SimParticlemv_compressDetStepMCs__NeutralsResampler.
   102588848         1125.0      0.336  mu2e::CrvSteps_compressDetStepMCs__NeutralsResampler.
     3906805           42.8      0.013  mu2e::GenParticles_compressDetStepMCs__NeutralsResampler.
     1659538           18.2      0.005  mu2e::StatusG4_g4run__NeutralsResampler.
     1257933           13.8      0.004  art::EventIDs_neutralsResampler__NeutralsResampler.
      999378           11.0      0.003  mu2e::StrawGasSteps_compressDetStepMCs__NeutralsResampler.
      862598            9.5      0.003  mu2e::CaloShowerSteps_compressDetStepMCs__NeutralsResampler.
      728436            8.0      0.002  EventAuxiliary
--------------------------------------
   305508504         3350.2      1.000  Total

Reply:
Suppose we split this into a separate data product:

1. What other parts of SimParticle, if any, should move with it?
2. Can we do it in a way that does not require maintenance on the compression routines or the stage>1 G4 code?

Reply:
We had a similar discussion about Bob's request to store information prior to the interaction for protons in the stopping target.  Does anyone remember what we did for that case?  I don't see it in SimParticle.hh .

Reply:
> We had a similar discussion about Bob's request to store information prior to the interaction for protons in the stopping target. Does anyone remember what we did for that case? I don't see it in SimParticle.hh .

From a thread 9/17/18, it looks like in Bob's case the variable "lastMomentum" was always zero, and we changed it to be the momentum at the last step before the interaction

Reply:
Note that the _preLastStepKE member is still taking space, even though it does not seem to be used:  the value is hardcoded to be -1 and there are no setters.   May be it is time to take it out, along with the accessor.

Reply:
As we discussed at the meeting I stored the pre-last step info in the previously existing data members;
BTW, the endMomentum is also based on the pre-last step info as is endKE, 
so it all may be actually more consistent now